data/scripts.js

```diff
@@ -472,85 +529,142 @@ let BattleScripts = {
 		move.totalDamage = 0;
-		/** @type {number | false | undefined} */
-		let damage = 0;
 		pokemon.lastDamage = 0;
-		if (move.multihit) {
-			let hits = move.multihit;
-			if (Array.isArray(hits)) {
-				// yes, it's hardcoded... meh
-				if (hits[0] === 2 && hits[1] === 5) {
-					if (this.gen >= 5) {
-						hits = this.sample([2, 2, 3, 3, 4, 5]);
-					} else {
-						hits = this.sample([2, 2, 2, 3, 3, 3, 4, 5]);
-					}
+		let hits = move.multihit || 1;
+		if (Array.isArray(hits)) {
+			// yes, it's hardcoded... meh
+			if (hits[0] === 2 && hits[1] === 5) {
+				if (this.gen >= 5) {
+					hits = this.sample([2, 2, 3, 3, 4, 5]);
 				} else {
-					hits = this.random(hits[0], hits[1] + 1);
+					hits = this.sample([2, 2, 2, 3, 3, 3, 4, 5]);
 				}
+			} else {
+				hits = this.random(hits[0], hits[1] + 1);
 			}
-			hits = Math.floor(hits);
-			let nullDamage = true;
-			/** @type {number | false | undefined} */
-			let moveDamage;
-			// There is no need to recursively check the ´sleepUsable´ flag as Sleep Talk can only be used while asleep.
-			let isSleepUsable = move.sleepUsable || this.getMove(move.sourceEffect).sleepUsable;
-			let i;
-			for (i = 0; i < hits && target.hp && pokemon.hp; i++) {
-				if (pokemon.status === 'slp' && !isSleepUsable) break;
-				move.hit = i + 1;
-
-				if (move.multiaccuracy && i > 0) {
-					accuracy = move.accuracy;
-					if (accuracy !== true) {
-						if (!move.ignoreAccuracy) {
-							boosts = this.runEvent('ModifyBoost', pokemon, null, null, Object.assign({}, pokemon.boosts));
-							boost = this.clampIntRange(boosts['accuracy'], -6, 6);
-							if (boost > 0) {
-								accuracy *= boostTable[boost];
-							} else {
-								accuracy /= boostTable[-boost];
-							}
-						}
-						if (!move.ignoreEvasion) {
-							boosts = this.runEvent('ModifyBoost', target, null, null, Object.assign({}, target.boosts));
-							boost = this.clampIntRange(boosts['evasion'], -6, 6);
-							if (boost > 0) {
-								accuracy /= boostTable[boost];
-							} else if (boost < 0) {
-								accuracy *= boostTable[-boost];
-							}
+		}
+		hits = Math.floor(hits);
+		let nullDamage = true;
+		/** @type {(number | boolean | undefined)[]} */
+		let moveDamage;
+		// There is no need to recursively check the ´sleepUsable´ flag as Sleep Talk can only be used while asleep.
+		let isSleepUsable = move.sleepUsable || this.getMove(move.sourceEffect).sleepUsable;
+
+		/** @type {(Pokemon | false | null)[]} */
+		let targetsCopy = targets.slice(0);
+		let i;
+		for (i = 0; i < hits && !damage.includes(false); i++) {
+			if (pokemon.status === 'slp' && !isSleepUsable) break;
+			move.hit = i + 1;
+			targetsCopy = targets.slice(0);
+			let target = targetsCopy[0]; // some relevant-to-single-target-moves-only things are hardcoded
+
+			// like this (Triple Kick)
+			if (target && move.multiaccuracy && i > 0) {
+				let accuracy = move.accuracy;
+				let boostTable = [1, 4 / 3, 5 / 3, 2, 7 / 3, 8 / 3, 3];
```

Comment:
seems like there should be a canonical place for accuracy and stat boost tables.